q2cli/handlers.py

```diff
@@ -188,43 +192,65 @@ def get_value(self, arguments, fallback=None):
 def parameter_handler_factory(name, semtype, default=NoDefault):
     ast = semtype.to_ast()
     if ast['name'] == 'Metadata':
-        return MetadataHandler(name)
+        return MetadataHandler(name, default=default)
     elif ast['name'] == 'MetadataCategory':
-        return MetadataCategoryHandler(name)
-    return RegularParameterHandler(name, semtype, default=default)
+        return MetadataCategoryHandler(name, default=default)
+    else:
+        return RegularParameterHandler(name, semtype, default=default)
 
 
 class MetadataHandler(Handler):
-    def __init__(self, name):
-        super().__init__(name, prefix='m_')
+    def __init__(self, name, default=NoDefault):
+        super().__init__(name, prefix='m_', default=default)
         self.click_name += '_file'
 
     def get_click_options(self):
-        yield click.Option(['--' + self.cli_name],
-                           type=click.Path(exists=True, dir_okay=False),
-                           help='Metadata mapping file')
+        name = '--' + self.cli_name
+        type = click.Path(exists=True, dir_okay=False)
+        help = 'Metadata mapping file'
+
+        # Metadata currently supports a default of None. Anything else makes it
+        # required.
+        if self.default is None:
+            yield click.Option([name], type=type, default=self.default,
+                               help='%s  [optional]' % help)
+        else:
+            yield click.Option([name], type=type, help='%s  [required]' % help)
 
     def get_value(self, arguments, fallback=None):
         path = self._locate_value(arguments, fallback)
         return qiime.Metadata.load(path)
 
 
 class MetadataCategoryHandler(Handler):
-    def __init__(self, name):
-        super().__init__(name, prefix='m_')
+    def __init__(self, name, default=NoDefault):
+        super().__init__(name, prefix='m_', default=default)
         self.name = name
         self.click_names = ['m_%s_file' % name, 'm_%s_category' % name]
         self.cli_names = [
             q2cli.util.to_cli_name(n) for n in self.click_names]
 
     def get_click_options(self):
-        yield click.Option(['--' + self.cli_names[0]],
-                           type=click.Path(exists=True, dir_okay=False),
-                           help='Metadata mapping file')
-
-        yield click.Option(['--' + self.cli_names[1]],
-                           type=str,
-                           help='Category from metadata mapping file')
+        md_name = '--' + self.cli_names[0]
+        md_type = click.Path(exists=True, dir_okay=False)
+        md_help = 'Metadata mapping file'
+
+        mdc_name = '--' + self.cli_names[1]
+        mdc_type = str
+        mdc_help = 'Category from metadata mapping file'
```

Comment:
This is fine, but an alternative would be to `**kwarg` in these arguments, then you would only have to deal with the presence/absence of `default` key and appending `[required]/[optional]` to `help`.